include/swift/AST/Stmt.h

```diff
@@ -991,6 +1057,14 @@ class CaseStmt final : public Stmt,
   }
 
   static bool classof(const Stmt *S) { return S->getKind() == StmtKind::Case; }
+
+  size_t numTrailingObjects(OverloadToken<CaseLabelItem>) const {
+    return getNumCaseLabelItems();
+  }
+
+  size_t numTrailingObjects(OverloadToken<FallthroughStmt *>) const {
+    return size_t(hasFallthrough);
```

Comment:
This is a bit *too* clever, I'd suggest to use clear syntax like `hasFallthrough ? 1 : 0`